### DISCUSSION TITLE
Support `testcode` directive for docstring formatting

### DISCUSSION THREAD

**Discussion by harupy** [astral-sh/ruff#9871] [Q&A]
Ruff supports `code-block` and `sourcecode` directives. Is it possible to support `testcode` directive?

https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html#directive-testcode

**Comment by MichaReiser** [comment]
@BurntSushi what are your thoughts on adding support for `testcode`?

**Comment by harupy** [comment]
@MichaReiser Thanks for the comment! We decided to remove `testcode` in https://github.com/mlflow/mlflow and use `code-block` instead :)
